File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
   
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int>m;
        vector<int>ans;
        for(int i = 0; i < nums.size(); i++){
            if(m.find(target-nums[i]) != m.end()){
                ans.push_back(i);
                ans.push_back(m[target-nums[i]]);
            }else{
                m[nums[i]] = i;
            }
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let map = new Map();
    let ans = [];
    nums.map((num,index) => {
        if(map.has(target - num)){
            ans = [index,map.get(target-num)];
        }else{
            map.set(num,index);
        }
    })
    return ans;
};